enhancements/network/bgp-ovn-kubernetes.md

```diff
@@ -0,0 +1,366 @@
+API. With BGP, this is no longer necessary. A user can simply configure RouteAdvertisements, and the pod subnet routes
+will be dynamically learned by an external gateway capable of BGP peering.
+
+##### Egress IP
+
+EgressIP feature that is dynamically moved between nodes. By enabling the feature in RouteAdvertisements, OVN-Kubernetes
+will automatically change frr-k8s configuration so that the node where an egress IP resides will advertise the IP. This
+eliminates the need for nodes to be on the same layer 2 segment, as we no longer have to rely on gratuitous ARP (GARP).
+
+##### Services
+
+LoadBalancer and External IP are supported by MetalLB. This feature will handle advertising cluster IPs for services.
+Note, the usefulness of advertising ClusterIP may be limited as it will be advertised by multiple nodes (similar to
+anycast) and may not work well with stateful connections.
+
+##### Egress Service
+
+Full support.
+
+##### Egress Firewall
+
+Full support.
+
+##### Egress QoS
+
+Full Support.
+
+
+##### Network Policy/ANP
+
+Full Support.
+
+##### Direct Pod Ingress
+
+Direct pod ingress is already enabled for the default cluster network. With direct pod ingress, any external entity can
+talk to a pod IP if it sends the packet to the node where the pod lives. Previously, support for direct pod ingress on
+user-defined networks was not supported. RouteAdvertisements with this enhancement may select user-defined networks and
+enable pod network advertisements. Therefore, it only makes sense to also accept direct pod ingress for these
+user-defined networks as well, especially since we know the selected subnets will not overlap in IP address space.
+
+Modifications will need to be made so that for packets arriving on br-ex, flows are added to steer the selected pod
+subnets to the right OVN GR patch port. Additionally, modifications will need to be made in the kernel routing table to
+leak routes to the pod subnets from each user-defined VRF into the default VRF routing table.
+
+#### Deployment Considerations
+
+##### MetalLB
+
+MetalLB is supported today as a day 2 add-on operator. MetalLB supports FRR mode, and has recently added support for
+FRR-K8S mode. When MetalLB is deployed in FRR-K8S mode, it handles instantiating FRR-K8S. For the purposes of
+OVN-Kubernetes integration, we need this as a day 0 function. For pods relying on the underlay and not using an overlay
+encapsulation, the BGP routes will need to be learned by the time the pods come up.
+
+In order to achieve day 0 functionality, FRR-K8S will be launched by CNO as a host networked pod on each node (where
+applicable). MetalLB will be modified so that it can launch in FRR-K8S mode, without deploying FRR-K8S, and integrate
+with the one launched by CNO.
+
+##### Bare Metal
+
+Bare Metal deployments expose a virtual IP (VIP) using keepalived. This VIP is moved between control-plane nodes, and
```

Comment:
if this is related to the cluster api, we'd need frr-k8s to be up (and configure) even before CNO starts. Also, keepalived works with ARP while here we assume the interaction with a router. All in all I'd leave this out for now (maybe even as a non goal)

Reply:
@cybertron would like to get your thoughts on the keepalived stuff.

Reply:
I need to look a little deeper on whether this would work, but I can say that currently our answer to cross-subnet deployments is that you disable keepalived and use an external LB of some sort. I have a few thoughts about how this _might_ work, but at this point I would say we don't _need_ it. My guess would be that most users deploying an advanced configuration like this will already have standalone loadbalancers they'd prefer to use anyway. If that proves not to be the case we can always revisit integrating keepalived with the BGP framework.

Reply:
Thanks Ben. As we discussed will remove support for this for now, and list is as a potential future goal.